netket/sampler/metropolis.py

```diff
@@ -153,7 +217,8 @@ class MetropolisSampler(Sampler):
 
     The dtype of the sampled states can be chosen.
     """
-
+    n_chains_per_rank: int = struct.field(pytree_node=False, default=None, repr=False)
```

Comment:
I moved it, because I could not see a reason for it to be in the base class anymore, and it simplified things as the number of chains computation was split between the two . See the commit 06a18733329a67c0954d87c69ad0147fdfdce6ff

Reply:
I'm ok with moving that logic to Metropolis. In the end, that is the only sampler that uses multiple chains..